src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDao.java

```diff
@@ -561,8 +561,10 @@ FireStoreDirectoryEntry computeDirectory(
 
     for (FireStoreDirectoryEntry dirItem : enumComputed) {
       totalSize = totalSize + dirItem.getSize();
-      crc32cCollection.add(StringUtils.lowerCase(dirItem.getChecksumCrc32c()));
-      if (dirItem.getChecksumMd5() != null) {
+      if (!StringUtils.isEmpty(dirItem.getChecksumCrc32c())) {
```

Comment:
Whenever I see `StringUtils.isEmpty()` used, I wonder if this is another way of writing `!= null` or if there's really a case where this field contains the empty string. If it's checking for `null`, using an explicit check would make the code's intent clearer.

Reply:
I disagree.  I think in both cases we want to ignore these entries

Reply:
I'm not saying we shouldn't handle both cases, I'm asking if both cases can actually arise in practice. I think we should understand our code well enough to be able to answer this question.

From looking at the code, I can see how the field could be `null`, but I don't see how the field could be the empty string.

Reply:
The problem it's not just a matter of our code. I've found variability in the different cloud apis (with little documentation on their end)

Reply:
Do you mean cloud APIs in general or in this specific case? It looks to me like these fields are set using our code but maybe I'm missing something.